hrabsence/CRM/HRAbsence/BAO/HRAbsencePeriod.php

```diff
@@ -134,4 +136,21 @@ public static function getAbsencePeriods($startDate = NULL, $endDate = NULL) {
 
     return $data;
   }
+
+  /**
+   * Add a new task to the Entitlement Recalculation queue to recalculate it for
+   * the given absence period
+   *
+   * @param \CRM_HRAbsence_BAO_HRAbsencePeriod $period
```

Comment:
I prefer to pass objects as much as possible (we're doing OOP after all, aren't we?). This has a couple of advantages:
- If we need to change the method to work based on more absence period fields, then we won't need to update the method signature
- We can use IDE autocomplete feature

Also, MAYBE, there is a bit of performance improvement as objects are passed as a reference, whereas the id would be passed by value, resulting a new copy of the variable (maybe the copy-on-write feature helps with that, but I don't know tbh)